lamindb/models/_from_values.py

```diff
@@ -22,56 +22,55 @@ def get_or_create_records(
     field: StrField,
     *,
     create: bool = False,
-    from_source: bool = False,
     organism: Record | str | None = None,
     source: Record | None = None,
     mute: bool = False,
+    _from_source: bool = False,
 ) -> RecordList:
     """Get or create records from iterables."""
     from .query_set import RecordList
 
     registry = field.field.model  # type: ignore
+    if isinstance(field, str):
+        field = registry._meta.get_field(field)
+    organism_record = _get_organism_record(field, organism, values=iterable)
+    # TODO: the create is problematic if field is not a name field
     if create:
-        return RecordList([registry(**{field.field.name: value}) for value in iterable])  # type: ignore
-    organism = _get_organism_record(field, organism)
+        create_kwargs = {}
+        if organism_record:
+            create_kwargs["organism"] = organism_record
+        return RecordList(
+            [
+                registry(**{field.field.name: value}, **create_kwargs)
+                for value in iterable
+            ]
+        )  # type: ignore
+
     iterable_idx = index_iterable(iterable)
 
     # returns existing records & non-existing values
     records, nonexist_values, msg = get_existing_records(
         iterable_idx=iterable_idx,
         field=field,
-        organism=organism,
+        organism=organism_record,
         mute=mute,
     )
 
     # new records to be created based on new values
     if len(nonexist_values) > 0:
-        source_record = None
-        if from_source:
-            if isinstance(source, Record):
-                source_record = source
-        if not source_record and hasattr(registry, "public"):
-            if organism is None:
-                organism = _ensembl_prefix(nonexist_values[0], field, organism)
-                organism = _get_organism_record(field, organism, force=True)
-
-        if source_record:
-            from bionty.core._add_ontology import check_source_in_db
-
-            check_source_in_db(registry=registry, source=source_record)
-
-            from_source = not source_record.in_db
-        elif hasattr(registry, "source_id"):
-            from_source = True
-        else:
-            from_source = False
-
-        if from_source:
+        if _from_source and hasattr(registry, "source_id"):
+            # if can and needed, get organism record from the existing records
+            if (
+                organism_record is None
+                and len(records) > 0
+                and _require_organism(registry)
```

Comment:
Can you rename this function to something that makes clear what it does? Like `check_if_organism_is_required()`?